pykafka/producer.py

```diff
@@ -540,7 +575,40 @@ def flush(self, linger_ms, release_pending=False):
         """
         self._wait_for_flush_ready(linger_ms)
         with self.lock:
-            batch = [self.queue.pop() for _ in range(len(self.queue))]
+            batch = []
+            batch_size_in_bytes = 0
+            while len(self.queue) > 0:
+                # TODO: Should we optimistically pop and renenqueue vs. peak
+                peaked_message = self.queue[-1]
+
+                if peaked_message and peaked_message.value is not None:
+                    if len(peaked_message.value) > max_request_size:
```

Comment:
You're better off using `len(peeked_message)` here - that also takes into account the length of the key and other overhead.